### PR TITLE
Expose `AudioSink::empty()`

### PR DESCRIPTION
**Pull request by hankjordan**
# Objective

Exposes `empty()` method for `AudioSink`.
Based on `0.10.0`, should be a non-breaking change.

---

## Changelog

- Expose `empty()` method for `AudioSink`
- Add `AudioSink::empty()` example

### CODE REVIEW

File: examples/audio/audio_empty.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+use bevy::prelude::*;
```
Comment (cart) [inline examples/audio/audio_empty.rs:1]: Definitely a matter of preference, but I don't think we need an example just to illustrate the empty method. With that criteria we'd have thousands of examples.

**Review by cart — COMMENTED** [review]
I rebased on main to fix the diff (it was including changelog and publish changes)

File: examples/audio/audio_empty.rs (hunk 1 above)
Comment (mockersf) [reply examples/audio/audio_empty.rs:1]: You could update one of the existing audio example, like audio_control, to exit once the sink is empty

Comment (cart) [reply examples/audio/audio_empty.rs]: I just think its solving a problem that doesn't need to be solved. I think we should wait until the demand for such an example presents itself.

**Pull request merged**